### PR TITLE
Include build from source instructions

### PR DESCRIPTION
**Pull request by bhatpmk**
This PR includes BUILD_FROM_SOURCE_README.md, with the instructions to build Keycloak 20.0.1 image from source.

### CODE REVIEW

File: BUILD_FROM_SOURCE_README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+The base tag this release is branched from is `20.0.1`
+
+---
+## Build
+Prerequisites: Building Keycloak from source requires the following:
+* JDK 11 (or newer)
+* Maven 3.5.4 (or newer).
+
+Building Keycloak server from source and creating an image, assumes the source files from the repository https://github.com/verrazzano/keycloak from branch oracle/release/20.0.1 or a branch created from oracle/release/20.0.1, is available under base directory _verrazzano_.
```
Comment (lake-of-dreams) [inline BUILD_FROM_SOURCE_README.md]: Maybe just mention in steps
1. Clone https://github.com/verrazzano/keycloak
2. checkout branch oracle/release/20.0.1 or a branch created from oracle/release/20.0.1
3. etc..

File: BUILD_FROM_SOURCE_README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# Build Instructions
+
+The base tag this release is branched from is `20.0.1`
```
Comment (lake-of-dreams) [inline BUILD_FROM_SOURCE_README.md]: Maybe a link to the tag is more useful

File: BUILD_FROM_SOURCE_README.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+Building Keycloak server from source and creating an image, assumes the source files from the repository https://github.com/verrazzano/keycloak from branch oracle/release/20.0.1 or a branch created from oracle/release/20.0.1, is available under base directory _verrazzano_.
+
+To build only the Keycloak server, run:  
+```
+ cd verrazzano/keycloak/quarkus
+ # build the project for the first time to put required modules of Keycloak into local maven cache in package org.keycloak
+ mvn --settings ../maven-settings.xml -f ../pom.xml clean install -DskipTestsuite -DskipExamples -DskipTests
+ 
+ # build Keycloak Quarkus distribution
+ mvn --settings ../maven-settings.xml -f dist/pom.xml clean install
```
Comment (lake-of-dreams) [inline BUILD_FROM_SOURCE_README.md:21]: why is this a two step process, I mean should only one mvn invocation be not sufficient to put in local cache and as well as build it

File: BUILD_FROM_SOURCE_README.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+```
+
+A successful maven build creates the distribution archive keycloak-20.0.1.tar.gz under keycloak/quarkus/dist/target.  
+
+Copy keycloak-<version>.tar.gz and other files required for building the image to verrazzano/keycloak-containers/server.  
+```
+ cd verrazzano/keycloak
+ cp quarkus/dist/target/keycloak-20.0.1.tar.gz keycloak/quarkus/container
```
Comment (lake-of-dreams) [inline BUILD_FROM_SOURCE_README.md]: so it copies from `verrazzano/keycloak/quarkus/dist/target/keycloak-20.0.1.tar.gz` to `verrazzano/keycloak/keycloak/quarkus/container` ? some other path is mentioned in line above 
`verrazzano/keycloak-containers/server`

File: BUILD_FROM_SOURCE_README.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+In order to tag and push the docker image, define the following environment variables:
+
+```
+ export DOCKER_REPO=<Docker Repository>
+ export DOCKER_NAMESPACE=<Docker Namespace>
+ export DOCKER_TAG=<Docker Tag>
+```
+
+Build and push the image to container registry:
+```
```
Comment (lake-of-dreams) [inline BUILD_FROM_SOURCE_README.md:48]: the line above defines `DOCKER_TAG` but in this line it uses `DOCKER_IMAGE_NAME` and `DOCKER_IMAGE_TAG`

File: BUILD_FROM_SOURCE_README.md (hunk 3 above)
Comment (bhatpmk) [reply BUILD_FROM_SOURCE_README.md:21]: I have followed the instruction from https://github.com/keycloak/keycloak/tree/20.0.1/quarkus. I can investigate this further separately.

File: BUILD_FROM_SOURCE_README.md (hunk 5 above)
Comment (bhatpmk) [reply BUILD_FROM_SOURCE_README.md:48]: Fixed the instruction

File: BUILD_FROM_SOURCE_README.md (hunk 4 above)
Comment (bhatpmk) [reply BUILD_FROM_SOURCE_README.md]: Updated the instruction

File: BUILD_FROM_SOURCE_README.md (hunk 2 above)
Comment (bhatpmk) [reply BUILD_FROM_SOURCE_README.md]: Done

File: BUILD_FROM_SOURCE_README.md (hunk 1 above)
Comment (bhatpmk) [reply BUILD_FROM_SOURCE_README.md]: Done

**Review by lake-of-dreams — APPROVED** [review]

**Pull request merged**